app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
-
+    people = {}
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+    for person in people:
+        name = person["name"]
+        age = person["age"]
+        person_instance = Person(name, age)
+        result.append(person_instance)
+    for person in people:
+        person_instance = Person.people.get(person["name"])
+        if person_instance:
+            if person.get("wife"):
+                wife_name = person["wife"]
+                person_instance.wife = Person.people.get(wife_name)
+            elif person.get("husband"):
+                husband_name = person["husband"]
+                person_instance.husband = Person.people.get(husband_name)
```

Comment:
The code assumes that a person can have either a 'wife' or a 'husband', but not both. This might not cover all possible cases. Additionally, if a person is both a 'wife' and a 'husband' in different contexts, the last assignment will overwrite the previous one. Consider handling these relationships more flexibly.